src/pykeen/utils.py

```diff
@@ -931,6 +933,43 @@ def extend_batch(
     return hrt_batch
 
 
+def check_shapes(
+    *x: Tuple[Union[torch.Tensor, Tuple[int, ...]], str],
+    raise_on_errors: bool = True,
+) -> bool:
+    """Verify that a sequence of tensors are of matching shapes.
+
+    :param x:
+        A tuple (t, s), where `t` is a tensor, or an actual shape of a tensor (a tuple of integers), and `s` is a
+        string, where each character corresponds to a (named) dimension. If the shapes of different tensors share a
+        character, the corresponding dimensions are expected to be of equal size.
+    :param raise_on_errors:
```

Comment:
Maybe add a concrete example?

Reply:
done in 22bf857177d3ea03f51c4e325e78a5e84e604d8f